common/src/compile.rs

```diff
@@ -365,13 +366,45 @@ pub async fn compile_from_source(
 /// Creates a [Project] from an Etherscan source.
 pub fn etherscan_project(metadata: &Metadata, target_path: impl AsRef<Path>) -> Result<Project> {
     let target_path = dunce::canonicalize(target_path.as_ref())?;
+    let sources_path = target_path.join(&metadata.contract_name);
     metadata.source_tree().write_to(&target_path)?;
 
-    let paths = ProjectPathsConfig::builder().build_with_root(target_path);
+    let mut settings = metadata.source_code.settings()?.unwrap_or_default();
+
+    // make remappings absolute
+    for remapping in settings.remappings.iter_mut() {
+        let mut path = Path::new(&remapping.path);
+        if path.is_absolute() {
+            path = path.strip_prefix("/").unwrap_or(path);
+        }
+        let new_path = target_path.join(path);
+        remapping.path = new_path.display().to_string();
+    }
+
+    // add missing remappings
+    let oz = Remapping {
+        name: "@openzeppelin/".into(),
+        path: sources_path.join("@openzeppelin").display().to_string(),
+    };
+    settings.remappings.push(oz);
```

Comment:
is there an argument for always adding oz?

Reply:
it's pretty common and etherscan itself doesn't include it, see example in PR description